modules/telemetry/events/settings-events.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace VIPWorkflow\Modules\Telemetry\Events;
+
+use Automattic\VIP\Telemetry\Tracks;
+use VIPWorkflow\VIP_Workflow;
+
+class Settings_Events {
+	/**
+	 * Tracks instance
+	 *
+	 * @var Tracks
+	 */
+	protected Tracks $tracks;
+
+	/**
+	 * Constructor
+	 *
+	 * @param Tracks $tracks The Tracks instance
+	 */
+	public function __construct( Tracks $tracks ) {
+		$this->tracks = $tracks;
+	}
+
+	/**
+	 * Record an event when the plugin is upgraded
+	 *
+	 * @param string $previous_version The previous version
+	 * @param string $new_version The new version
+	 */
+	public function record_admin_update( string $previous_version, string $new_version ): void {
+		// Get all custom statuses
+		$custom_statuses = VIP_Workflow::instance()->custom_status->get_custom_statuses();
+		// Get supported post types
+		$supported_post_types = VIP_Workflow::instance()->custom_status->get_supported_post_types();
+
+		$total_posts = 0;
+		foreach ( $supported_post_types as $post_type ) {
+			// Get all posts count for each post type
+			$posts_count = wp_count_posts( $post_type );
+
+			// Only care about published and posts with custom status
+			$total_posts += (int) $posts_count->publish;
```

Comment:
@hanifn Still curious about this data point [RE this reply](https://github.com/Automattic/vip-workflow-plugin/pull/21#discussion_r1733529489) from last week.

My basic problem is that if we combine `publish` and custom status post counts together, the posts in custom statuses will likely be overwhelmed by published post counts, and then we'll essentially lose data about custom statuses. Why include the count of published posts at all? And if we want that data for a reason, how about also including `custom_status_posts` as a separate count?

Reply:
> statuses will likely be overwhelmed by published post counts, and then we'll essentially lose data about custom statuses. Why include the count of published posts at all? And if we want that data for a reason, how about also including custom_status_posts as a separate count?

@alecgeatches Yes I see your point. I think adding the `custom_status_posts` would also be useful. Would additional comparison points be more useful? I'm thinking adding `custom_status_posts` and `published_posts` on top of the existing `total_posts`

Reply:
If `total_posts === custom_status_posts + published_posts`, then we probably just need the two terms on the right side and I say we drop `total_posts`. But in general, if we're already doing the queries we need to count posts it doesn't really hurt to send more counts.

Reply:
Agreed. Changed in c9ac0647a7a79da1316e03f9d6eb5408ffdc094e

Reply:
It looks like we're not actually using `$custom_status_posts`, is that right?

Reply:
> It looks like we're not actually using `$custom_status_posts`, is that right?

Oops that's a mistake. Fixed 163daded7fb13b2408c548bda8c2543ffb2aa186